### PR TITLE
versions checks to avoid mixing major versions across the EasyBuild components

### PR DESCRIPTION
**Pull request by branfosj**
This PR will add various version checks and is EB5 prep for the EB4 branch.

* [x] check that framework and blocks major versions are the same and fails if not
* [x] check `--from-pr` / `--include-easyblocks-from-pr` branches match in use versions

### CODE REVIEW

File: easybuild/tools/version.py [hunk 1]
```diff
@@ -103,3 +104,16 @@ def this_is_easybuild():
...
+    """Compare major versions"""
+    # Deal with version instances being either strings or LooseVersion
+    if not isinstance(v1, LooseVersion):
+        v1 = LooseVersion(v1)
+    if not isinstance(v2, LooseVersion):
+        v2 = LooseVersion(v2)
+
+    if v1.version[0] == v2.version[0]:
+        return False
+    return True
```
Comment (boegel) [inline easybuild/tools/version.py:119]:
```suggestion
    return v1.version[0] != v2.version[0]

File: easybuild/main.py [hunk 2]
```diff
@@ -618,6 +619,13 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None, pr
     (build_specs, _log, logfile, robot_path, search_query, eb_tmpdir, try_to_generate,
      from_pr_list, tweaked_ecs_paths) = cfg_settings
 
+    # compare running Framework and EasyBlocks versions
+    if EASYBLOCKS_VERSION == UNKNOWN_VERSION:
+        print_msg("Unable to determine EasyBlocks version, so we'll assume it is not different from Framework")
```
Comment (boegel) [inline easybuild/main.py:624]: This should be a warning?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/main.py (hunk 2 above)
Comment (branfosj) [reply easybuild/main.py]: 0a5c99f

File: easybuild/tools/version.py (hunk 1 above)
Comment (branfosj) [reply easybuild/tools/version.py]: 0a5c99f

File: easybuild/main.py (hunk 2 above)
Comment (branfosj) [reply easybuild/main.py]: Printing something here is problematic, so we'll log it as a warning instead.

The most likely place for this to happen is running the framework unit tests without an easyblocks installation. If you try to use EB in that case then you'll fail really quickly.

The only other case I can think of is if something breaks the version lookup of the blocks. That is a rare enough possibility that I think we should ignore.

File: easybuild/tools/github.py [hunk 3]
```diff
@@ -587,9 +588,39 @@ def fetch_files_from_pr(pr, path=None, github_user=None, github_account=None, gi
         else:
             raise EasyBuildError("Couldn't find path to patched file %s", full_path)
 
+    if github_repo == GITHUB_EASYCONFIGS_REPO:
+        ver = _get_version_for_repo(os.path.join(final_path, 'setup.py'))
+    elif github_repo == GITHUB_EASYBLOCKS_REPO:
+        ver = _get_version_for_repo(os.path.join(final_path, 'easybuild', 'easyblocks', '__init__.py'))
+
+    if different_major_versions(FRAMEWORK_VERSION, ver):
+        raise EasyBuildError("Framework (%s) is a different major version than PR target (%s)." % (FRAMEWORK_VERSION,
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/tools/github.py:597]: fixed in https://github.com/branfosj/easybuild-framework/pull/10

**Review by boegel — CHANGES_REQUESTED** [review]
@branfosj see https://github.com/branfosj/easybuild-framework/pull/10

**Review by boegel — APPROVED** [review]

**Pull request merged**